src/target/drivers/input/rotary/rotary.c

```diff
@@ -14,26 +26,34 @@ static u32 last_rotary_clock = 0;
 
 void ROTARY_Init()
 {
-    rcc_peripheral_enable_clock(&RCC_APB2ENR, RCC_APB2ENR_IOPCEN);
+    rcc_periph_clock_enable(get_rcc_from_pin(ROTARY_PIN0));
+    rcc_periph_clock_enable(get_rcc_from_pin(ROTARY_PIN1));
+    rcc_periph_clock_enable(get_rcc_from_pin(ROTARY_PRESS_PIN));
```

Comment:
this is redundant? can we creates a set of macro to make hardware.h looks like a table close to what we have in our wiki?

and a code to scan the table to initialize IO.
and a header file to expose the right macros to the rest of code? like pagelist.h and page.h we are doing?

Reply:
you're right.  I have an assert to ensure they're the same already.  no need t do it 3 times